content/blog/secure-cloud-kubernetes-workloads-intro/index.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "Securely access cloud resources in Kubernetes workloads"
+date: 2020-10-13T20:56:15-07:00
+draft: true
+meta_desc: An introduction for running accessing cloud resources securely for Kubernetes workloads.
+meta_image: meta.png
+authors:
+    - lee-briggs
+tags:
+    - kubernetes
+    - security
+    - cloud-engineering
+---
+
+As you build your cloud-native Kubernetes applications, you might eventually find you need to access cloud resources that reside outside your Kubernetes cluster. Perhaps you need to store static files in an object store (Amazon S3, Google Cloud Storage or Azure Blog Storage) or use a queuing system to pass messages to other services (Amazon SQS, Azure Service Bus or Google Pub/Sub).
+
+Once you start to access these services from within your application, you'll need to find a way to authenticate to the cloud providers API, which involves somehow managing credentials. It can be tempting at this point to create a "user" in your cloud provider and then create credentials like AWS Access Keys or a Google Cloud Service Account Key and store those inside your Kubernetes workload. A common approach is to store these credentials as a Kubernetes secret and then consume that secret as an environment variable in your running pods:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-secret
+data:
+  username: bXktYXBw
+  password: Mzk1MjgkdmRnN0pi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: secret-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: nginx
+      volumeMounts:
+        # name must match the volume name below
+        - name: secret-volume
+          mountPath: /etc/secret-volume
+  # The secret data is exposed to Containers in the Pod through a Volume.
+  volumes:
+    - name: secret-volume
+      secret:
+        secretName: test-secret
+```
+
+This method works and will allow access to cloud resources from your app. You can approach this with a security-conscious mindset and make sure the permissions that the keys are associated with are narrowly scoped (following the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)). However, unfortunately, if you have an information security team looking over your shoulder, they will still see this practice and be extremely concerned.
+
+In this first in a series of posts, we'll talk about why this might be a problem, examine the mindset of a malicious user attacking your Kubernetes cluster and introduce the mechanisms for avoiding these patterns.
```

Comment:
```suggestion
In this first in a series of posts, we'll talk about why this might be a problem, examine the mindset of a malicious user attacking your Kubernetes cluster, and introduce the mechanisms for avoiding these patterns.
```